serverless/azure/src/azure_source.ts

```diff
@@ -0,0 +1,107 @@
+import { RangeResponse, Source } from "pmtiles";
+import { BlobClient, ContainerClient } from "@azure/storage-blob";
+import { DefaultAzureCredential } from "@azure/identity";
+
+function streamToBuffer(stream: NodeJS.ReadableStream) {
+  return new Promise<Uint8Array>((resolve, reject) => {
+    const chunks: Uint8Array[] = [];
+    stream.on("data", (data) => {
+      chunks.push(Buffer.isBuffer(data) ? data : Buffer.from(data));
+    });
+    stream.on("end", () => {
+      resolve(new Uint8Array(Buffer.concat(chunks)));
+    });
+    stream.on("error", reject);
+  });
+}
+
+function getAzureDetails(mapName: string) {
+  let defaultAccountName = process.env.AZURE_STORAGE_ACCOUNT_NAME;
+  let defaultContainerName = process.env.AZURE_STORAGE_CONTAINER_NAME;
+  let defaultBlobName = process.env.AZURE_STORAGE_BLOB_NAME;
```

Comment:
In the AWS and Cloudfront implementations there is a hard constraint of 1 function = 1 bucket. It looks like this behavior is something specific to your use case in Azure? I would strongly prefer there is as close to 100% parity between AWS/Cloudflare/Azure meaning there shouldn't be any way to override using `${mapName}` - was there a situation where the existing design did not meet your requirements?

Reply:
The reason I did that is that we need to specify a blob name, which is a path that could be `maps/basemaps/protomaps/20231124.pmtiles` for map name `default` and `maps/custom_maps/rail/europe_20231123.pmtiles` for map name `rail`. 

If we want the function to support multiple maps by `${mapName}` in the URL, I can't think of an easy way to map that. I would prefer if the Function App didn't decide how we organize our files, like it does in the AWS and Cloudflare cases. I guess we just make the `AZURE_STORAGE_BLOB_NAME` overridable with `${mapName}`, and let the users only specify a specific storage account and container. 

I don't have a strong opinion in this case, I'll let you decide the solution. We don't store custom maps in the same folder as basemaps and the files don't have the same name. But we use our own code that can support our needs.

Reply:
Sorry for the late reply (I'm also going offline for a week tomorrow but will prioritize this when I get back)

> I would prefer if the Function App didn't decide how we organize our files

Then the easiest thing to do is to map the bucket key directly to the URL. It looks like you want to perform "aliasing", where a presentation name like `map1` in the URL hits another key in the storage bucket. If we support this feature in Azure then we need to make it work in CF and AWS as well. This kind of aliasing could happen outside this code function logic in your own application-specific code.

Reply:
> Then the easiest thing to do is to map the bucket key directly to the URL. It looks like you want to perform "aliasing", where a presentation name like `map1` in the URL hits another key in the storage bucket. If we support this feature in Azure then we need to make it work in CF and AWS as well. This kind of aliasing could happen outside this code function logic in your own application-specific code.

Well, you could say that it is in the application code, because `getAzureStorageSource` is called in the application code and you can easily implement your own version, the source takes the details as paramerters. But I can change so that the default `getAzureStorageSource` doesn't support "aliasing" and you would have to implement your own version if you need more advanced logic. I'll also change so that it just replaces `{name}` in the blob name, just like the other sources.